public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -312,6 +312,11 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
       prevCondition && setValue('condition', prevCondition);
     } else {
       setValue('type', RuleFormType.cloudAlerting);
+      const newDsName = getDataSourceSrv().getInstanceSettings(queries[0].datasourceUid)?.name;
```

Comment:
i would add a comment in here to understand why we are only setting the data source in case of cloud